carbon-design-system/carbon
# fix: datepicker onchange trigger issue

Closes #16125 
Closes #13128 
Closes #15925
Closes #14047
Closes #14465   

When clicking outside a DatePickerInput (after having already set a date previously) the on Change handler fires with unexpected dates in the payload of the handler.

#### Changelog

**Changed**

- set the allowInput prop on DatePicker to false


#### Testing / Reviewing

- set two dates with the datepicker.

- click again on one of the inputs to open the calendar

- click outside or click on another form element or tab away, Dates should be unchanged .
